### ISSUE TITLE
VisualVM hangs on welcome screen

### ISSUE THREAD

**Issue by gowthamv** [oracle/visualvm#168]
I have downloaded visualVM 1.3.9 from https://visualvm.github.io/releases.html. 

receiving java.util.concurrent.TimeoutException error while lauching visualVM 1.3.9

**VisualVM log**

Compiler: HotSpot 64-Bit Tiered Compilers
Heap memory usage: initial 24.0MB maximum 228.0MB
Non heap memory usage: initial 2.4MB maximum -1b
Garbage collector: PS Scavenge (Collections=12 Total time spent=0s)
Garbage collector: PS MarkSweep (Collections=2 Total time spent=0s)
Classes: loaded=6339 total loaded=6339 unloaded 0
INFO [org.netbeans.core.ui.warmup.DiagnosticTask]: Total memory 8,364,048,384
INFO [null]: Total physical memory 8,364,048,384
INFO [org.netbeans.modules.autoupdate.updateprovider.DownloadListener]: Reading URL https://visualvm.github.io/uc/release139/updates.xml.gz?unique=09387b029-3b9e-4ba0-a6a9-b1a8259c1d04_eccbfec2-a008-4458-bf0c-2612de30f28e failed (java.io.IOException: Timeout while opening connection to https://visualvm.github.io/uc/release139/updates.xml.gz?unique=09387b029-3b9e-4ba0-a6a9-b1a8259c1d04_eccbfec2-a008-4458-bf0c-2612de30f28e)
INFO [org.netbeans.modules.autoupdate.ui.actions.AutoupdateCheckScheduler]
java.util.concurrent.TimeoutException
	at java.util.concurrent.FutureTask.get(FutureTask.java:205)
	at org.netbeans.modules.autoupdate.updateprovider.NetworkAccess$Task$1.run(NetworkAccess.java:109)
Caused: java.io.IOException: Timeout while opening connection to https://visualvm.github.io/uc/release139/updates.xml.gz?unique=09387b029-3b9e-4ba0-a6a9-b1a8259c1d04_eccbfec2-a008-4458-bf0c-2612de30f28e
	at org.netbeans.modules.autoupdate.updateprovider.NetworkAccess$Task$1.run(NetworkAccess.java:129)
Caused: java.io.IOException: Timeout while opening connection to https://visualvm.github.io/uc/release139/updates.xml.gz?unique=09387b029-3b9e-4ba0-a6a9-b1a8259c1d04_eccbfec2-a008-4458-bf0c-2612de30f28e
	at org.netbeans.modules.autoupdate.updateprovider.DownloadListener.notifyException(DownloadListener.java:103)
	at org.netbeans.modules.autoupdate.updateprovider.AutoupdateCatalogCache.copy(AutoupdateCatalogCache.java:246)
	at org.netbeans.modules.autoupdate.updateprovider.AutoupdateCatalogCache.writeCatalogToCache(AutoupdateCatalogCache.java:99)
	at org.netbeans.modules.autoupdate.updateprovider.AutoupdateCatalogProvider.refresh(AutoupdateCatalogProvider.java:154)
	at org.netbeans.modules.autoupdate.services.UpdateUnitProviderImpl.refresh(UpdateUnitProviderImpl.java:180)
	at org.netbeans.api.autoupdate.UpdateUnitProvider.refresh(UpdateUnitProvider.java:196)
[catch] at org.netbeans.modules.autoupdate.ui.actions.AutoupdateCheckScheduler$4.run(AutoupdateCheckScheduler.java:155)
	at org.openide.util.RequestProcessor$Task.run(RequestProcessor.java:1423)
	at org.openide.util.RequestProcessor$Processor.run(RequestProcessor.java:2033)

**Screenshots**

**Desktop (please complete the following information):**
Product Version         = VisualVM 1.3.9
  Operating System        = Windows 10 version 10.0 running on amd64
  Java; VM; Vendor        = 1.8.0_212; OpenJDK 64-Bit Server VM 25.212-b04; Amazon.com Inc.
  Runtime                 = OpenJDK Runtime Environment 1.8.0_212-b04

**Comment by jisedlac** [comment]
The TimeoutException is harmless, VisualVM just logs an unsuccessful attempt to access the Plugins Center (no network/invalid proxy configuration etc.).

What exactly does "hangs on welcome screen" mean? Starts up and the UI is not responsive (like no menu can be displayed)? Please provide more details and attach the full VisualVM logfile. Thanks!

**Comment by gowthamv** [comment]
Yes. It started up and  UI is not responsive , not able click any menus available in welcome screen.

**Comment by gowthamv** [comment]
Is it possible to org.netbeans.modules.autoupdate.ui.actions.AutoupdateCheckScheduler disable this option using  visualvm\etc\visualvm.conf file to resolve this issue?

**Comment by jisedlac** [comment]
As I've mentioned, the TimeoutException thrown in autoupdate code should be harmless. We need the full VisualVM logfile and ideally a thread dump from the stuck VisualVM instance to analyze the problem. You can use the '`jstack`' tool in your `JAVA_HOME\bin` folder to take the thread dump.

**Comment by gowthamv** [comment]
@jisedlac  PFB the VisualVM logfile 
-------------------------------------------------------------------------------
>Log Session: Saturday, January 11, 2020 8:04:09 PM IST
>System Info: 
  Product Version         = VisualVM 1.3.9
  Operating System        = Windows 10 version 10.0 running on amd64
  Java; VM; Vendor        = 1.8.0_212; OpenJDK 64-Bit Server VM 25.212-b04; Amazon.com Inc.
  Runtime                 = OpenJDK Runtime Environment 1.8.0_212-b04
  Java Home               = C:\JA\Java\jdk1.8.0_212_64VMJMC\jre
  System Locale; Encoding = en_US (visualvm); Cp1252
  Home Directory          = C:\Users\test
  Current Directory       = C:\Users\test\Downloads\vm\visualvm_139\bin
  User Directory          = C:\Users\test\AppData\Roaming\VisualVM\1.3.9
  Cache Directory         = C:\Users\test\AppData\Local\VisualVM\Cache\1.3.9
  Installation            = C:\Users\test\Downloads\vm\visualvm_139\visualvm
                            C:\Users\test\Downloads\vm\visualvm_139\profiler
                            C:\Users\test\Downloads\vm\visualvm_139\platform
  Boot & Ext. Classpath   = C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\lib\resources.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\lib\rt.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\lib\sunrsasign.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\lib\jsse.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\lib\jce.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\lib\charsets.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\lib\jfr.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\classes;C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\lib\ext\access-bridge-64.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\lib\ext\cldrdata.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\lib\ext\dnsns.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\lib\ext\jaccess.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\lib\ext\jfxrt.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\lib\ext\localedata.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\lib\ext\nashorn.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\lib\ext\sunec.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\lib\ext\sunjce_provider.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\lib\ext\sunmscapi.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\lib\ext\sunpkcs11.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\jre\lib\ext\zipfs.jar
  Application Classpath   = C:\Users\test\Downloads\vm\visualvm_139\platform\lib\boot.jar;C:\Users\test\Downloads\vm\visualvm_139\platform\lib\org-openide-modules.jar;C:\Users\test\Downloads\vm\visualvm_139\platform\lib\org-openide-util-lookup.jar;C:\Users\test\Downloads\vm\visualvm_139\platform\lib\org-openide-util.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\lib\dt.jar;C:\JA\Java\jdk1.8.0_212_64VMJMC\lib\tools.jar
  Startup Classpath       = C:\Users\test\Downloads\vm\visualvm_139\platform\core\core.jar;C:\Users\test\Downloads\vm\visualvm_139\platform\core\org-openide-filesystems.jar;C:\Users\test\Downloads\vm\visualvm_139\visualvm\core\com-sun-tools-visualvm-modules-startup.jar;C:\Users\test\Downloads\vm\visualvm_139\visualvm\core\locale\core_visualvm.jar
-------------------------------------------------------------------------------
INFO [org.netbeans.core.startup.NbEvents]: Turning on modules:
	org.openide.util.lookup [8.25.2 20161002-268c3bcd7a42]
	org.openide.util [8.39.1 20161002-268c3bcd7a42]
	org.openide.modules [7.43.1 20161002-268c3bcd7a42]
	org.netbeans.api.annotations.common/1 [1.24.1 20161002-268c3bcd7a42]
	org.openide.filesystems [8.12.1 20161002-268c3bcd7a42]
	org.openide.awt [7.62.1 20161002-268c3bcd7a42]
	org.netbeans.api.progress/1 [1.38.1 20161002-268c3bcd7a42]
	org.openide.dialogs [7.38.1 20161002-268c3bcd7a42]
	org.openide.nodes [7.39.1 20161002-268c3bcd7a42]
	org.openide.windows [6.71.1 20161002-268c3bcd7a42]
	org.netbeans.modules.editor.mimelookup/1 [1.36.1 20161002-268c3bcd7a42]
	org.openide.text [6.62.2 20161002-268c3bcd7a42]
	org.netbeans.swing.tabcontrol [1.51.1 20161002-268c3bcd7a42]
	org.netbeans.swing.outline [1.30.1 20161002-268c3bcd7a42]
	org.openide.explorer [6.57.1 20161002-268c3bcd7a42]
	org.openide.actions [6.35.1 20161002-268c3bcd7a42]
	org.netbeans.modules.queries/1 [1.39.1 20161002-268c3bcd7a42]
	org.openide.loaders [7.57.2 20161002-268c3bcd7a42]
	org.openide.io [1.45.1 20161002-268c3bcd7a42]
	org.netbeans.swing.plaf [1.37.1 20161002-268c3bcd7a42]
	org.netbeans.spi.quicksearch [1.23.1 20161002-268c3bcd7a42]
	org.netbeans.bootstrap/1 [2.69.3 20161002-268c3bcd7a42]
	org.netbeans.core.startup/1 [1.55.2 20161002-268c3bcd7a42]
	org.netbeans.modules.settings/1 [1.45.1 20161002-268c3bcd7a42]
	org.netbeans.modules.sendopts/2 [2.31.1 20161002-268c3bcd7a42]
	org.netbeans.modules.sampler [1.10.1 20161002-268c3bcd7a42]
	org.netbeans.modules.progress.ui [1.29.2 20161002-268c3bcd7a42]
	org.netbeans.modules.profiler.utilities/1 [1.31.1 20161002-268c3bcd7a42]
	org.netbeans.lib.profiler/1 [1.59.4 20161002-268c3bcd7a42]
	org.netbeans.lib.profiler.charts/1 [1.25.1 20161002-268c3bcd7a42]
	org.netbeans.lib.profiler.common/1 [1.32.1 20161002-268c3bcd7a42]
	org.netbeans.modules.profiler.api/1 [1.19.1 20161002-268c3bcd7a42]
	org.netbeans.lib.profiler.ui/1 [1.45.1 20161002-268c3bcd7a42]
	org.netbeans.modules.keyring [1.20.1 20161002-268c3bcd7a42]
	org.netbeans.core/2 [3.46.1 20161002-268c3bcd7a42]
	org.netbeans.modules.options.api/1 [1.41.1 20161002-268c3bcd7a42]
	org.netbeans.modules.profiler.selector.api/1 [1.29.1 20161002-268c3bcd7a42]
	org.netbeans.modules.profiler.attach/2 [2.15.1 20161002-268c3bcd7a42]
	org.netbeans.modules.profiler.selector.ui/1 [1.26.1 20161002-268c3bcd7a42]
	org.netbeans.modules.profiler/2 [2.22.1 20161002-268c3bcd7a42]
	org.netbeans.modules.profiler.snaptracer/1 [1.17.1 20161002-268c3bcd7a42]
	org.netbeans.modules.profiler.oql/2 [2.12.1 20161002-268c3bcd7a42]
	org.netbeans.modules.profiler.heapwalker [1.99.2 20161002-268c3bcd7a42]
	org.netbeans.libs.jna/1 [1.38.1 20161002-268c3bcd7a42]
	org.netbeans.modules.masterfs/2 [2.48.3 20161002-268c3bcd7a42]
	org.netbeans.modules.masterfs.windows [1.13.1 20161002-268c3bcd7a42]
	org.netbeans.modules.masterfs.nio2 [1.12.1 20161002-268c3bcd7a42]
	org.netbeans.modules.keyring.fallback [1.4.1 20161002-268c3bcd7a42]
	org.netbeans.modules.keyring.impl [1.20.1 20161002-268c3bcd7a42]
	org.netbeans.modules.editor.mimelookup.impl/1 [1.27.1 20161002-268c3bcd7a42]
	org.netbeans.modules.autoupdate.services [1.47.2 20161002-268c3bcd7a42]
	org.netbeans.modules.autoupdate.ui [1.41.1 20161002-268c3bcd7a42]
	org.netbeans.modules.autoupdate.cli [1.9.1 20161002-268c3bcd7a42]
	org.netbeans.libs.jna.platform/1 [1.8.1 20161002-268c3bcd7a42]
	org.netbeans.core.windows/2 [2.77.2 20161002-268c3bcd7a42]
	org.netbeans.core.ui/1 [1.41.1 20161002-268c3bcd7a42]
	org.netbeans.core.network [1.6.1 20161002-268c3bcd7a42]
	org.netbeans.core.nativeaccess/1 [1.27.1 20161002-268c3bcd7a42]
	org.netbeans.core.multitabs/1 [1.8.1.1 1 20161002-268c3bcd7a42]
	com.sun.tools.visualvm.uisupport/0 [1.3 161004]
	com.sun.tools.visualvm.core/0 [1.6 161004]
	com.sun.tools.visualvm.host/0 [1.0 161004]
	com.sun.tools.visualvm.application/0 [1.9 161004]
	com.sun.tools.visualvm.tools/0 [1.4 161004]
	com.sun.tools.visualvm.coredump/0 [1.1 161004]
	com.sun.tools.visualvm.threaddump/0 [1.0 161004]
	com.sun.tools.visualvm.heapdump/0 [1.1 161004]
	com.sun.tools.visualvm.profiling/0 [1.3 161004]
	com.sun.tools.visualvm.sampler [2.1 161004]
	com.sun.tools.visualvm.sa/0 [1.1 161004]
	com.sun.tools.visualvm.modules.startup [1.1 161004]
	com.sun.tools.visualvm.profiler/0 [1.5 161004]
	com.sun.tools.visualvm.modules.appui [1.0 161004]
	com.sun.tools.visualvm.jvmstat/0 [1.0 161004]
	com.sun.tools.visualvm.jvm/0 [1.1 161004]
	com.sun.tools.visualvm.jmx/0 [1.4 161004]
	com.sun.tools.visualvm.charts/0 [1.5 161004]
	com.sun.tools.visualvm.host.views/0 [1.0 161004]
	com.sun.tools.visualvm.host.remote/0 [1.0 161004]
	com.sun.tools.visualvm.attach/0 [1.1 161004]
	com.sun.tools.visualvm.application.views/0 [1.0 161004]

INFO [org.netbeans.ui.metrics.laf]: USG_LOOK_AND_FEEL
WARNING [org.openide.filesystems.Ordering]: Not all children in Windows2/Modes/explorer/ marked with the position attribute: [ExplorerTopComponent.wstcref], but some are: [profiler_cp.wstcref]
Diagnostic information
Input arguments:
	-Xms24m
	-Xmx256m
	-Dnetbeans.accept_license_class=com.sun.tools.visualvm.modules.startup.AcceptLicense
	-Dsun.jvmstat.perdata.syncWaitMs=10000
	-Dsun.java2d.noddraw=true
	-Dsun.java2d.d3d=false
	--add-exports=java.desktop/sun.awt=ALL-UNNAMED
	--add-exports=jdk.jvmstat/sun.jvmstat.monitor.event=ALL-UNNAMED
	--add-exports=jdk.jvmstat/sun.jvmstat.monitor=ALL-UNNAMED
	--add-exports=java.desktop/sun.swing=ALL-UNNAMED
	--add-exports=jdk.attach/sun.tools.attach=ALL-UNNAMED
	--add-modules=java.activation
	-XX:+IgnoreUnrecognizedVMOptions
	-Djdk.home=C:\JA\Java\jdk1.8.0_212_64VMJMC
	-Dnetbeans.home=C:\Users\test\Downloads\vm\visualvm_139\platform
	-Dnetbeans.user=C:\Users\test\AppData\Roaming\VisualVM\1.3.9
	-Dnetbeans.default_userdir_root=C:\Users\test\AppData\Roaming\VisualVM
	-XX:+HeapDumpOnOutOfMemoryError
	-XX:HeapDumpPath=C:\Users\test\AppData\Roaming\VisualVM\1.3.9\var\log\heapdump.hprof
	-Dsun.awt.keepWorkingSetOnMinimize=true
	-Dnetbeans.dirs=C:\Users\test\Downloads\vm\visualvm_139\visualvm;C:\Users\test\Downloads\vm\visualvm_139\profiler
Compiler: HotSpot 64-Bit Tiered Compilers
Heap memory usage: initial 24.0MB maximum 228.0MB
Non heap memory usage: initial 2.4MB maximum -1b
Garbage collector: PS Scavenge (Collections=12 Total time spent=0s)
Garbage collector: PS MarkSweep (Collections=2 Total time spent=0s)
Classes: loaded=6352 total loaded=6352 unloaded 0
INFO [org.netbeans.core.ui.warmup.DiagnosticTask]: Total memory 8,364,048,384
INFO [null]: Total physical memory 8,364,048,384
INFO [org.netbeans.modules.autoupdate.updateprovider.DownloadListener]: Reading URL https://visualvm.github.io/uc/release139/updates.xml.gz?unique=0348e090b-7e84-44e2-aab2-088e623f3780_754cc6cc-d17a-4435-b443-e1c1e6f5f5b3 failed (java.io.IOException: Timeout while opening connection to https://visualvm.github.io/uc/release139/updates.xml.gz?unique=0348e090b-7e84-44e2-aab2-088e623f3780_754cc6cc-d17a-4435-b443-e1c1e6f5f5b3)
INFO [org.netbeans.modules.autoupdate.ui.actions.AutoupdateCheckScheduler]
java.util.concurrent.TimeoutException
	at java.util.concurrent.FutureTask.get(FutureTask.java:205)
	at org.netbeans.modules.autoupdate.updateprovider.NetworkAccess$Task$1.run(NetworkAccess.java:109)
Caused: java.io.IOException: Timeout while opening connection to https://visualvm.github.io/uc/release139/updates.xml.gz?unique=0348e090b-7e84-44e2-aab2-088e623f3780_754cc6cc-d17a-4435-b443-e1c1e6f5f5b3
	at org.netbeans.modules.autoupdate.updateprovider.NetworkAccess$Task$1.run(NetworkAccess.java:129)
Caused: java.io.IOException: Timeout while opening connection to https://visualvm.github.io/uc/release139/updates.xml.gz?unique=0348e090b-7e84-44e2-aab2-088e623f3780_754cc6cc-d17a-4435-b443-e1c1e6f5f5b3
	at org.netbeans.modules.autoupdate.updateprovider.DownloadListener.notifyException(DownloadListener.java:103)
	at org.netbeans.modules.autoupdate.updateprovider.AutoupdateCatalogCache.copy(AutoupdateCatalogCache.java:246)
	at org.netbeans.modules.autoupdate.updateprovider.AutoupdateCatalogCache.writeCatalogToCache(AutoupdateCatalogCache.java:99)
	at org.netbeans.modules.autoupdate.updateprovider.AutoupdateCatalogProvider.refresh(AutoupdateCatalogProvider.java:154)
	at org.netbeans.modules.autoupdate.services.UpdateUnitProviderImpl.refresh(UpdateUnitProviderImpl.java:180)
	at org.netbeans.api.autoupdate.UpdateUnitProvider.refresh(UpdateUnitProvider.java:196)
[catch] at org.netbeans.modules.autoupdate.ui.actions.AutoupdateCheckScheduler$4.run(AutoupdateCheckScheduler.java:155)
	at org.openide.util.RequestProcessor$Task.run(RequestProcessor.java:1423)
	at org.openide.util.RequestProcessor$Processor.run(RequestProcessor.java:2033)

**PFB the jstack log.**

2020-01-11 20:09:13
Full thread dump OpenJDK 64-Bit Server VM (25.212-b04 mixed mode):

"pool-4-thread-1" #38 prio=5 os_prio=0 tid=0x0000000018ebc800 nid=0x1918 in Object.wait() [0x000000001eaeb000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at org.netbeans.core.windows.services.DialogDisplayerImpl.notify(DialogDisplayerImpl.java:297)
	- locked <0x00000000faf5ad60> (a org.netbeans.core.windows.services.DialogDisplayerImpl$1AWTQuery)
	at org.netbeans.core.windows.services.DialogDisplayerImpl.notify(DialogDisplayerImpl.java:173)
	at org.netbeans.core.NbAuthenticator.getPasswordAuthentication(NbAuthenticator.java:101)
	at java.net.Authenticator.requestPasswordAuthentication(Authenticator.java:317)
	- locked <0x00000000f0b3bc68> (a org.netbeans.core.NbAuthenticator)
	at sun.net.www.protocol.http.spnego.NegotiateCallbackHandler.getAnswer(NegotiateCallbackHandler.java:68)
	at sun.net.www.protocol.http.spnego.NegotiateCallbackHandler.handle(NegotiateCallbackHandler.java:90)
	at com.sun.security.auth.module.Krb5LoginModule.promptForName(Krb5LoginModule.java:858)
	at com.sun.security.auth.module.Krb5LoginModule.attemptAuthentication(Krb5LoginModule.java:704)
	at com.sun.security.auth.module.Krb5LoginModule.login(Krb5LoginModule.java:617)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at javax.security.auth.login.LoginContext.invoke(LoginContext.java:755)
	at javax.security.auth.login.LoginContext.access$000(LoginContext.java:195)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:682)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:680)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:680)
	at javax.security.auth.login.LoginContext.login(LoginContext.java:587)
	at sun.security.jgss.GSSUtil.login(GSSUtil.java:258)
	at sun.security.jgss.krb5.Krb5Util.getTicket(Krb5Util.java:158)
	at sun.security.jgss.krb5.Krb5InitCredential$1.run(Krb5InitCredential.java:337)
	at sun.security.jgss.krb5.Krb5InitCredential$1.run(Krb5InitCredential.java:333)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.security.jgss.krb5.Krb5InitCredential.getTgt(Krb5InitCredential.java:332)
	at sun.security.jgss.krb5.Krb5InitCredential.getInstance(Krb5InitCredential.java:145)
	at sun.security.jgss.krb5.Krb5MechFactory.getCredentialElement(Krb5MechFactory.java:122)
	at sun.security.jgss.krb5.Krb5MechFactory.getMechanismContext(Krb5MechFactory.java:187)
	at sun.security.jgss.GSSManagerImpl.getMechanismContext(GSSManagerImpl.java:224)
	at sun.security.jgss.GSSContextImpl.initSecContext(GSSContextImpl.java:212)
	at sun.security.jgss.GSSContextImpl.initSecContext(GSSContextImpl.java:179)
	at sun.security.jgss.spnego.SpNegoContext.GSS_initSecContext(SpNegoContext.java:882)
	at sun.security.jgss.spnego.SpNegoContext.initSecContext(SpNegoContext.java:317)
	at sun.security.jgss.GSSContextImpl.initSecContext(GSSContextImpl.java:248)
	at sun.security.jgss.GSSContextImpl.initSecContext(GSSContextImpl.java:179)
	at sun.net.www.protocol.http.spnego.NegotiatorImpl.init(NegotiatorImpl.java:108)
	at sun.net.www.protocol.http.spnego.NegotiatorImpl.<init>(NegotiatorImpl.java:117)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at sun.net.www.protocol.http.Negotiator.getNegotiator(Negotiator.java:63)
	at sun.net.www.protocol.http.NegotiateAuthentication.isSupportedImpl(NegotiateAuthentication.java:138)
	- locked <0x00000000fae70728> (a java.lang.Class for sun.net.www.protocol.http.NegotiateAuthentication)
	at sun.net.www.protocol.http.NegotiateAuthentication.isSupported(NegotiateAuthentication.java:111)
	- locked <0x00000000f04fdb38> (a org.netbeans.ModuleManager$SystemClassLoader)
	at sun.net.www.protocol.http.AuthenticationHeader.parse(AuthenticationHeader.java:200)
	at sun.net.www.protocol.http.AuthenticationHeader.<init>(AuthenticationHeader.java:143)
	at sun.net.www.protocol.http.HttpURLConnection.doTunneling(HttpURLConnection.java:2097)
	- locked <0x00000000fad2f888> (a sun.net.www.protocol.https.DelegateHttpsURLConnection)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:183)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.connect(HttpsURLConnectionImpl.java:162)
	at org.netbeans.modules.autoupdate.updateprovider.NetworkAccess$Task$2.call(NetworkAccess.java:166)
	at org.netbeans.modules.autoupdate.updateprovider.NetworkAccess$Task$2.call(NetworkAccess.java:148)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

   Locked ownable synchronizers:
	- <0x00000000fe0e8b10> (a java.util.concurrent.ThreadPoolExecutor$Worker)

"pool-3-thread-1" #37 prio=5 os_prio=0 tid=0x0000000018ebb800 nid=0xbd8 waiting on condition [0x000000001e7ee000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000000fde7b4c8> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

   Locked ownable synchronizers:
	- None

"Timer-0" #35 daemon prio=1 os_prio=-2 tid=0x0000000018eb7000 nid=0x2910 in Object.wait() [0x000000001e5ef000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.util.TimerThread.mainLoop(Timer.java:552)
	- locked <0x00000000fdef2790> (a java.util.TaskQueue)
	at java.util.TimerThread.run(Timer.java:505)

   Locked ownable synchronizers:
	- None

"DestroyJavaVM" #27 prio=5 os_prio=0 tid=0x0000000018ebd000 nid=0x1ac8 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"Thread-2" #26 daemon prio=5 os_prio=0 tid=0x0000000019e7b000 nid=0x2e08 in Object.wait() [0x000000001cf1f000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.lang.Object.wait(Object.java:502)
	at java.util.prefs.AbstractPreferences$EventDispatchThread.run(AbstractPreferences.java:1482)
	- locked <0x00000000f069a230> (a java.util.LinkedList)

   Locked ownable synchronizers:
	- None

"AutoUpdate-UI" #25 daemon prio=1 os_prio=-2 tid=0x0000000019e7c800 nid=0x3618 waiting for monitor entry [0x000000001cc1e000]
   java.lang.Thread.State: BLOCKED (on object monitor)
	at org.netbeans.ModuleManager$SystemClassLoader.getResourcesImpl(ModuleManager.java:700)
	- waiting to lock <0x00000000f04fdb38> (a org.netbeans.ModuleManager$SystemClassLoader)
	at org.netbeans.ProxyClassLoader.getResources(ProxyClassLoader.java:390)
	at org.openide.util.lookup.MetaInfServicesLookup.search(MetaInfServicesLookup.java:199)
	at org.openide.util.lookup.MetaInfServicesLookup.beforeLookup(MetaInfServicesLookup.java:156)
	at org.openide.util.lookup.AbstractLookup.lookupItem(AbstractLookup.java:427)
	at org.openide.util.lookup.AbstractLookup.lookup(AbstractLookup.java:421)
	at org.openide.util.lookup.ProxyLookup.lookup(ProxyLookup.java:222)
	at org.netbeans.modules.autoupdate.updateprovider.InstalledUpdateProvider.defaultProvider(InstalledUpdateProvider.java:86)
	at org.netbeans.modules.autoupdate.updateprovider.InstalledUpdateProvider.access$000(InstalledUpdateProvider.java:70)
	at org.netbeans.modules.autoupdate.updateprovider.InstalledUpdateProvider$UP.getUpdateItems(InstalledUpdateProvider.java:128)
	at org.netbeans.modules.autoupdate.services.UpdateUnitFactory.getUpdateUnits(UpdateUnitFactory.java:127)
	at org.netbeans.modules.autoupdate.services.UpdateManagerImpl$Cache.<init>(UpdateManagerImpl.java:271)
	at org.netbeans.modules.autoupdate.services.UpdateManagerImpl.getCache(UpdateManagerImpl.java:231)
	at org.netbeans.modules.autoupdate.services.UpdateManagerImpl.getUpdateUnits(UpdateManagerImpl.java:100)
	at org.netbeans.api.autoupdate.UpdateManager.getUpdateUnits(UpdateManager.java:103)
	at org.netbeans.modules.autoupdate.ui.actions.AutoupdateCheckScheduler.checkUpdateElements(AutoupdateCheckScheduler.java:218)
	at org.netbeans.modules.autoupdate.ui.actions.AutoupdateCheckScheduler$5.run(AutoupdateCheckScheduler.java:189)
	at org.openide.util.RequestProcessor$Task.run(RequestProcessor.java:1423)
	at org.openide.util.RequestProcessor$Processor.run(RequestProcessor.java:2033)

   Locked ownable synchronizers:
	- None

"File Watcher" #24 daemon prio=1 os_prio=-2 tid=0x0000000019e7a800 nid=0x23f4 waiting on condition [0x000000001c7ee000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000000f06d8178> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
	at org.netbeans.modules.masterfs.watcher.windows.WindowsNotifier.nextEvent(WindowsNotifier.java:282)
	at org.netbeans.modules.masterfs.providers.Notifier$1.nextEvent(Notifier.java:119)
	at org.netbeans.modules.masterfs.watcher.Watcher$Ext.run(Watcher.java:329)
	at java.lang.Thread.run(Thread.java:748)

   Locked ownable synchronizers:
	- None

"TimerQueue" #22 daemon prio=5 os_prio=0 tid=0x00000000154e5800 nid=0x2708 waiting on condition [0x0000000016b0e000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000000f04efa70> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
	at java.util.concurrent.DelayQueue.take(DelayQueue.java:211)
	at javax.swing.TimerQueue.run(TimerQueue.java:174)
	at java.lang.Thread.run(Thread.java:748)

   Locked ownable synchronizers:
	- <0x00000000f015e108> (a java.util.concurrent.locks.ReentrantLock$NonfairSync)

"AWT-EventQueue-0" #21 prio=6 os_prio=0 tid=0x000000001543d800 nid=0x1bdc waiting for monitor entry [0x0000000016a0d000]
   java.lang.Thread.State: BLOCKED (on object monitor)
	at org.netbeans.ModuleManager$SystemClassLoader.getResourcesImpl(ModuleManager.java:700)
	- waiting to lock <0x00000000f04fdb38> (a org.netbeans.ModuleManager$SystemClassLoader)
	at org.netbeans.ProxyClassLoader.getResources(ProxyClassLoader.java:390)
	at org.openide.util.lookup.MetaInfServicesLookup.search(MetaInfServicesLookup.java:199)
	at org.openide.util.lookup.MetaInfServicesLookup.beforeLookup(MetaInfServicesLookup.java:156)
	at org.openide.util.lookup.MetaInfServicesLookup.beforeLookupResult(MetaInfServicesLookup.java:135)
	at org.openide.util.lookup.AbstractLookup.lookup(AbstractLookup.java:483)
	at org.openide.util.lookup.ProxyLookup$R.initResults(ProxyLookup.java:390)
	at org.openide.util.lookup.ProxyLookup$R.myBeforeLookup(ProxyLookup.java:669)
	at org.openide.util.lookup.ProxyLookup$R.computeResult(ProxyLookup.java:549)
	at org.openide.util.lookup.ProxyLookup$R.allInstances(ProxyLookup.java:509)
	at org.openide.util.lookup.ProxyLookup$R.allInstances(ProxyLookup.java:505)
	at org.openide.util.Lookup.lookupAll(Lookup.java:271)
	at org.netbeans.core.windows.services.DialogDisplayerImpl.customizeDlg(DialogDisplayerImpl.java:332)
	at org.netbeans.core.windows.services.DialogDisplayerImpl.access$000(DialogDisplayerImpl.java:76)
	at org.netbeans.core.windows.services.DialogDisplayerImpl$1AWTQuery.showDialog(DialogDisplayerImpl.java:262)
	at org.netbeans.core.windows.services.DialogDisplayerImpl$1AWTQuery.run(DialogDisplayerImpl.java:192)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

   Locked ownable synchronizers:
	- None

"AWT-Windows" #19 daemon prio=6 os_prio=0 tid=0x0000000015426000 nid=0x1184 runnable [0x000000001680e000]
   java.lang.Thread.State: RUNNABLE
	at sun.awt.windows.WToolkit.eventLoop(Native Method)
	at sun.awt.windows.WToolkit.run(WToolkit.java:316)
	at java.lang.Thread.run(Thread.java:748)

   Locked ownable synchronizers:
	- None

"AWT-Shutdown" #18 prio=5 os_prio=0 tid=0x0000000015425800 nid=0x34d0 in Object.wait() [0x000000001670f000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.lang.Object.wait(Object.java:502)
	at sun.awt.AWTAutoShutdown.run(AWTAutoShutdown.java:295)
	- locked <0x00000000f0400bc8> (a java.lang.Object)
	at java.lang.Thread.run(Thread.java:748)

   Locked ownable synchronizers:
	- None

"Java2D Disposer" #17 daemon prio=10 os_prio=2 tid=0x0000000015424800 nid=0x1a28 in Object.wait() [0x000000001656e000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:144)
	- locked <0x00000000f0480580> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:165)
	at sun.java2d.Disposer.run(Disposer.java:148)
	at java.lang.Thread.run(Thread.java:748)

   Locked ownable synchronizers:
	- None

"RequestProcessor queue manager" #16 daemon prio=5 os_prio=0 tid=0x000000001541b000 nid=0x6fc in Object.wait() [0x000000001646f000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at org.openide.util.RequestProcessor$TickTac.obtainFirst(RequestProcessor.java:2256)
	- locked <0x00000000f046bb70> (a java.lang.Class for org.openide.util.RequestProcessor$TickTac)
	at org.openide.util.RequestProcessor$TickTac.run(RequestProcessor.java:2232)

   Locked ownable synchronizers:
	- None

"CLI Requests Server" #13 daemon prio=5 os_prio=0 tid=0x0000000015320800 nid=0x3198 runnable [0x0000000015cef000]
   java.lang.Thread.State: RUNNABLE
	at java.net.DualStackPlainSocketImpl.accept0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketAccept(DualStackPlainSocketImpl.java:131)
	at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:409)
	at java.net.PlainSocketImpl.accept(PlainSocketImpl.java:199)
	- locked <0x00000000f0469e90> (a java.net.SocksSocketImpl)
	at java.net.ServerSocket.implAccept(ServerSocket.java:545)
	at java.net.ServerSocket.accept(ServerSocket.java:513)
	at org.netbeans.CLIHandler$Server.run(CLIHandler.java:1100)

   Locked ownable synchronizers:
	- None

"Active Reference Queue Daemon" #12 daemon prio=1 os_prio=-2 tid=0x00000000153a5800 nid=0x4e8 in Object.wait() [0x0000000015bee000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.lang.Object.wait(Object.java:502)
	at org.openide.util.lookup.implspi.ActiveQueue.removeBetter(ActiveQueue.java:225)
	at org.openide.util.lookup.implspi.ActiveQueue.access$400(ActiveQueue.java:57)
	at org.openide.util.lookup.implspi.ActiveQueue$Daemon.run(ActiveQueue.java:180)
	- locked <0x00000000f04a04e0> (a java.lang.ref.ReferenceQueue$Lock)

   Locked ownable synchronizers:
	- None

"Service Thread" #9 daemon prio=9 os_prio=0 tid=0x00000000133ce800 nid=0x3bb0 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"C1 CompilerThread2" #8 daemon prio=9 os_prio=2 tid=0x000000001337c800 nid=0x3b60 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"C2 CompilerThread1" #7 daemon prio=9 os_prio=2 tid=0x0000000013376000 nid=0x381c waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"C2 CompilerThread0" #6 daemon prio=9 os_prio=2 tid=0x000000001336e800 nid=0x2d00 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"Attach Listener" #5 daemon prio=5 os_prio=2 tid=0x000000001336c000 nid=0x3838 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"Signal Dispatcher" #4 daemon prio=9 os_prio=2 tid=0x000000001336b000 nid=0x1b3c runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
	- None

"Finalizer" #3 daemon prio=8 os_prio=1 tid=0x000000000292d800 nid=0x3220 in Object.wait() [0x00000000146ee000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:144)
	- locked <0x00000000f0391fd8> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:165)
	at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:216)

   Locked ownable synchronizers:
	- None

"Reference Handler" #2 daemon prio=10 os_prio=2 tid=0x0000000013358000 nid=0x73c in Object.wait() [0x00000000145ef000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	at java.lang.Object.wait(Object.java:502)
	at java.lang.ref.Reference.tryHandlePending(Reference.java:191)
	- locked <0x00000000f04cf4d8> (a java.lang.ref.Reference$Lock)
	at java.lang.ref.Reference$ReferenceHandler.run(Reference.java:153)

   Locked ownable synchronizers:
	- None

"VM Thread" os_prio=2 tid=0x0000000013335000 nid=0x3548 runnable 

"GC task thread#0 (ParallelGC)" os_prio=0 tid=0x0000000002848800 nid=0x2e48 runnable 

"GC task thread#1 (ParallelGC)" os_prio=0 tid=0x000000000284b800 nid=0x128c runnable 

"GC task thread#2 (ParallelGC)" os_prio=0 tid=0x000000000284d000 nid=0x3ab8 runnable 

"GC task thread#3 (ParallelGC)" os_prio=0 tid=0x000000000284e800 nid=0x2ad4 runnable 

"VM Periodic Task Thread" os_prio=2 tid=0x0000000014d1a000 nid=0x17ec waiting on condition 

JNI global references: 2161

**Comment by thurka** [comment]
This is duplicate of GH-48. See this [comment](https://github.com/oracle/visualvm/issues/48#issuecomment-352675816) in GH-48 for more details.
